tests/System.IO.Pipelines.Tests/PipelineReaderWriterFacts.cs

```diff
@@ -645,6 +645,64 @@ public void FlushAsync_ReturnsCompletedTaskWhenMaxSizeIfZero()
             Assert.True(flushTask.IsCompleted);
         }
 
+        [Fact]
+        public void TryRead_ShouldReturnTrueOnCommit()
```

Comment:
Add tests for exceptional situations please:
1. TryRead + TryRead 
2. TryRead + ReadAsync
3. ReadAsync + TryRead